packages/server/lib/clients/oauth1.client.ts

```diff
@@ -76,44 +72,37 @@ export class OAuth1Client {
     }
 
     async getOAuthAccessToken(oauth_token: string, oauth_token_secret: string, oauth_token_verifier: string): Promise<any> {
-        let additionalTokenParams = {};
+        let additionalTokenParams: Record<string, any> = {};
         if (this.authConfig.token_params) {
             additionalTokenParams = this.authConfig.token_params;
         }
 
-        const promise = new Promise<any>((resolve, reject) => {
+        const promise = new Promise<Record<string, any>>((resolve, reject) => {
             // This is lifted from https://github.com/ciaranj/node-oauth/blob/master/lib/oauth.js#L456
             // Unfortunately that main method does not expose extra params like the initial token request does ¯\_(ツ)_/¯
 
-            // @ts-expect-error
             additionalTokenParams['oauth_verifier'] = oauth_token_verifier;
 
-            // @ts-expect-error
+            // @ts-expect-error we access private method
             this.client._performSecureRequest(
                 oauth_token,
                 oauth_token_secret,
-                // @ts-expect-error
+                // @ts-expect-error we access private method
                 this.client._clientOptions.accessTokenHttpMethod,
-                // @ts-expect-error
+                // @ts-expect-error we access private method
                 this.client._accessUrl,
                 additionalTokenParams,
                 null,
                 undefined,
-                // @ts-expect-error
-                function (error, data, response) {
-                    if (error) reject(error);
-                    else {
-                        // @ts-expect-error
-                        const queryParams = new URLSearchParams(data);
-
-                        const parsedFull = {};
-                        for (const pair of queryParams) {
-                            // @ts-expect-error
-                            parsedFull[pair[0]] = pair[1];
-                        }
-
-                        resolve(parsedFull);
+                function (error, data, _response) {
+                    if (error) {
+                        reject(error);
+                        return;
                     }
+
+                    const queryParams = new URLSearchParams(typeof data === 'string' ? data : data?.toString());
+
+                    resolve(Object.fromEntries(queryParams));
```

Comment:
Can you double check if I fixed correctly? ☺️

Reply:
can we add tests to validate it with code?

Reply:
I can extract this logic yes